harness/src/utils.rs

```diff
@@ -128,8 +93,81 @@ pub fn kill_process_using_port(port: u16) {
         .expect("Failed to execute kill command");
 }
 
+pub async fn collect_state_matching_rafts_until_leader_emerge(
+    rafts: &HashMap<u64, Raft>,
+    target_role: StateRole,
+) -> HashSet<u64> {
+    let mut target_set = HashSet::with_capacity(rafts.len());
+    let mut leader_merged = false;
+    let mut ex_term = 1;
+    // ! This loop can cause infinite hang
```

Comment:
Comments like these would be better placed as function's documentation rather than inline within the code.

Reply:
Thanks!